### ISSUE TITLE
SELECT DISTINCT is Broken

### ISSUE THREAD

**Issue by apavlo** [cmu-db/peloton#444]
We support the `DISTINCT` keyword in the parser but it looks like it doesn't do anything. It fails for both `INTEGER` and `VARCHAR` fields.

```
postgres=# CREATE TABLE foo (id INT PRIMARY KEY, name VARCHAR(32), val INT);
postgres=# INSERT INTO foo VALUES (1, 'BARBARBAR', 999);
INSERT INTO foo VALUES (1, 'BARBARBAR', 999) 1
postgres=# INSERT INTO foo VALUES (2, 'BARBARBAR', 999);
INSERT INTO foo VALUES (2, 'BARBARBAR', 999) 1
postgres=# INSERT INTO foo VALUES (3, 'ABLEABLEABLE', 888);
INSERT INTO foo VALUES (3, 'ABLEABLEABLE', 888) 1
postgres=# INSERT INTO foo VALUES (4, 'ABLEABLEABLE', 888);
INSERT INTO foo VALUES (4, 'ABLEABLEABLE', 888) 1

postgres=# SELECT * FROM foo;
 id |     name     | val 
----+--------------+-----
  1 | BARBARBAR    | 999
  2 | BARBARBAR    | 999
  3 | ABLEABLEABLE | 888
  4 | ABLEABLEABLE | 888
(4 rows)

postgres=# SELECT DISTINCT name FROM foo;
     name     
--------------
 BARBARBAR
 BARBARBAR
 ABLEABLEABLE
 ABLEABLEABLE
(4 rows)

postgres=# SELECT DISTINCT val FROM foo;
 val 
-----
 999
 999
 888
 888
(4 rows)
```

**Comment by jarulraj** [comment]
Hi @PauloAmora -- It would be great if you can add support for `DISTINCT`. You can try to use the `hash executor` functionality (https://github.com/cmu-db/peloton/blob/master/src/include/executor/hash_executor.h) to retrieve all the distinct entries in the hash table.

**Comment by PauloAmora** [comment]
I'll take a look into it.

**Comment by PauloAmora** [comment]
Hello @jarulraj,

So, I've started working on this on tuesday and tried to work out the logic of `HashPlan` and `HashExecutor`. Today, I started writing a new executor, using the output from the `HashExecutor` in a manner inspired by what `HashJoinPlan` does. Is this really needed or there would be a more elegant solution?

**Comment by PauloAmora** [comment]
Sure, but wouldn't this affect `HashJoin` as well?

**Comment by PauloAmora** [comment]
After a little more thinking, I guess I found an elegant way to solve this and don't affect `JOIN` queries.
By setting the duplicate results invisible (using `RemoveVisibility()`), the output works fine. However, in the `LogicalTile` implementation, there is a `GetValue()` deprecated implementation which asserts on tuple visibility, and I commented out the assertion for it to work. My question is: Is this acceptable?
